programs/drift/src/controller/orders.rs

```diff
@@ -2844,7 +2844,21 @@ pub fn fill_spot_order(
         false
     };
 
-    if should_cancel_market_order || should_cancel_reduce_only {
+    let should_cancel_for_no_borrow_liquidity = if is_open {
+        let market_index = user.orders[order_index].market_index;
+        let base_market = spot_market_map.get_ref(&market_index)?;
+        let quote_market = spot_market_map.get_quote_spot_market()?;
+        let (max_base_asset_amount, max_quote_asset_amount) =
+            get_max_fill_amounts(user, order_index, &base_market, &quote_market)?;
```

Comment:
why would we not want to keep the order open so when borrow liquidity comes back in the market the rest of the order can be filled?

Reply:
then a users orders will be stuck and they won't know why it's not filling, what do you think @0xbigz